### PR TITLE
added name and callback

### PR DESCRIPTION
**Pull request by sg3des**
added **name** for rigidbody - I think, name is important value, and place it in _userdata_ incorrectly.
added **callback** for rigidbody - if object collided and function callback is set, call it and send a pair of objects. 

what do you think?

example:
```go
collide := func(bodies [2]*tornago.RigidBody) {
	fmt.Println(bodies[0].Name, bodies[1].Name)
}

boxBody := tornago.NewRigidBody("boxBody")
boxBody.SetCallback(collide)
...
```

### CODE REVIEW

File: rigidbody.go [hunk 1]
```diff
@@ -90,8 +97,12 @@ type RigidBody struct {
 }
 
 // NewRigidBody returns a new rigid body with some default values.
-func NewRigidBody() *RigidBody {
+func NewRigidBody(name ...string) *RigidBody {
```
Comment (hydroflame) [inline rigidbody.go]: why variadic ?

File: rigidbody.go [hunk 2]
```diff
@@ -110,6 +121,11 @@ func (b *RigidBody) UserData() interface{} {
 	return b.userData
 }
 
+// SetCallback set this rigid body callback function where collided.
+func (b *RigidBody) SetCallback(callback func(bodies [2]*RigidBody)) {
```
Comment (hydroflame) [inline rigidbody.go]: Callback would also want the contact data

**Comment by hydroflame** [comment]
We could probably just make the callback field exported and not need methods

File: contactresolver.go [hunk 3]
```diff
@@ -30,12 +30,19 @@ func (c ContactResolver) ResolveContacts(contacts []Contact, duration float32) {
 	// calculate derivate data
 	derivateData := make([]contactDerivateData, len(contacts))
 	for x := 0; x < len(contacts); x++ {
+
+		//call callback function for each bodies
+		for _, b := range contacts[x].bodies {
```
Comment (hydroflame) [inline contactresolver.go]: unwrapped, sure

**Comment by hydroflame** [comment]
Also whats the point of naming your collision objects, I understand that it could be to have a unique identifier in callbacks but it seems that with closures that's irrelevant, thoughts ?

File: rigidbody.go (hunk 1 above)
Comment (sg3des) [inline rigidbody.go]: it does not matter, just do not want to break backward compatibility.
and to be able to do not specify the name, and I do not want to create a method SetName

**Comment by sg3des** [comment]
> Also whats the point of naming your collision objects, I understand that it could be to have a unique identifier in callbacks but it seems that with closures that's irrelevant, thoughts ?
yea...

> We could probably just make the callback field exported and not need methods
thx, i will think about it

**Comment by hydroflame** [comment]
I've added this to master but I'm not sure about names yet so I'm gonna close this :\

**Pull request closed without merging**